### ISSUE TITLE
Incompatible with git config core.commentChar = auto

### ISSUE THREAD

**Issue by SimonSapin** [MitMaro/git-interactive-rebase-tool#183]
The `core.commentChar` git configuration has a special value, `auto` which automatically selects a comment character based on the content of thing being edited:

https://git-scm.com/docs/git-config#Documentation/git-config.txt-corecommentChar

Setting this config breaks this tool:

```
hint: Waiting for your editor to close the file... Error reading file, Invalid action: #
error: There was a problem with the editor 'interactive-rebase-tool'.

**Comment by MitMaro** [comment]
Thanks for the report!

That's going to be a tricky one to handle.

**Comment by SimonSapin** [comment]
As a work around I’ve switched to `git config --global core.commentChar '%'`. This is less flexible, but still allows starting lines with `#` for Markdown headings.

**Comment by naokiri** [comment]
https://github.com/git/git/blob/117ddefdb4dfd9b40ae60967a7327754d8ce7a87/builtin/commit.c#L799-L800
https://github.com/git/git/blob/117ddefdb4dfd9b40ae60967a7327754d8ce7a87/builtin/commit.c#L631

According to git's code, git itself tries to find from only limited charset: `#;@!$%^&|:`

**Comment by SimonSapin** [comment]
But this tool does not care not about commit messages, right? Only interactive rebase commands files. Don’t non-comment lines in those always start with a letter? If so `auto` would be effectively equivalent to `#`.

**Comment by naokiri** [comment]
@SimonSapin You're right. I missed that point. So the better implementation was to check if it starts from '#' when comment_char is "auto". Let me update the PR now.

**Comment by MitMaro** [comment]
Great point @SimonSapin !

Thanks for the PR @naokiri ! I will get it merged now.

**Comment by MitMaro** [comment]
Fixed in #184
